NuRadioMC/simulation/simulation.py

```diff
@@ -1225,24 +1165,19 @@ def _create_meta_output_datastructures(self):
             self._output_maximum_amplitudes_envelope[station_id] = []
 
     def _create_station_output_structure(self, n_showers, n_antennas):
-        nS = 2 + 4 * self._n_reflections  # number of possible ray-tracing solutions
+        nS = self._raytracer.get_number_of_raytracing_solutions()  # number of possible ray-tracing solutions
         sg = {}
         sg['triggered'] = np.zeros(n_showers, dtype=np.bool)
         sg['shower_id'] = np.zeros(n_showers, dtype=np.int) * -1  # we need the reference to the shower id to be able to find the correct shower in the upper level hdf5 file
         sg['launch_vectors'] = np.zeros((n_showers, n_antennas, nS, 3)) * np.nan
         sg['receive_vectors'] = np.zeros((n_showers, n_antennas, nS, 3)) * np.nan
-        sg['ray_tracing_C0'] = np.zeros((n_showers, n_antennas, nS)) * np.nan
-        sg['ray_tracing_C1'] = np.zeros((n_showers, n_antennas, nS)) * np.nan
-        sg['ray_tracing_reflection'] = np.ones((n_showers, n_antennas, nS), dtype=np.int) * -1
-        sg['ray_tracing_reflection_case'] = np.ones((n_showers, n_antennas, nS), dtype=np.int) * -1
-        sg['ray_tracing_solution_type'] = np.ones((n_showers, n_antennas, nS), dtype=np.int) * -1
         sg['polarization'] = np.zeros((n_showers, n_antennas, nS, 3)) * np.nan
         sg['travel_times'] = np.zeros((n_showers, n_antennas, nS)) * np.nan
         sg['travel_distances'] = np.zeros((n_showers, n_antennas, nS)) * np.nan
-        sg['focusing_factor'] = np.ones((n_showers, n_antennas, nS))
         if(self._cfg['speedup']['amp_per_ray_solution']):
             sg['max_amp_shower_and_ray'] = np.zeros((n_showers, n_antennas, nS))
             sg['time_shower_and_ray'] = np.zeros((n_showers, n_antennas, nS))
+        self._raytracer.create_output_data_structure(sg, n_showers, n_antennas)
```

Comment:
the ray tracer should return a list of parameters. But not write them directly into the hdf5 file.

Reply:
Then how should we handle initial values? Does that also need to be specified or can we just initialize them all as nan or something like that?

Reply:
I would initialize everything as nan. Then it is clear if a value was filled of was just left empty

Reply:
Done. I added a parameter to specify the number of entries, in case we have multi-dimensional parameters